### ISSUE TITLE
Allow continuation after comments

### ISSUE THREAD

**Issue by legendarysrk** [docker/docker#29060]
This fix is to address #29005 where continuation after comments are not allowed after #24725.

This fix made the change so that #29005 could be addressed.

An integration test has been added. Other related tests have been udpated to conform to the behavior changes.

Please feel free for any suggestions.

This fix fixes #29005.

/cc @tianon @duglin @thaJeztah

**Comment by thaJeztah** [comment]
Duplicate of https://github.com/docker/docker/pull/29006
